src/utah-design-system/Button.jsx

```diff
@@ -0,0 +1,133 @@
+import clsx from 'clsx';
+import PropTypes from 'prop-types';
+import Spinner from './Spinner';
+import { createKeyLookup } from './utils';
+
+const COLORS = {
+  none: {
+    outlined: [
+      'border-slate-600 text-slate-600',
+      'focus:border-slate-600 focus:ring-slate-600',
+      'hover:border-slate-600 hover:bg-slate-600',
+    ].join(' '),
+    solid: [
+      'border-slate-600 bg-slate-600 text-white',
+      'focus:border-slate-600 focus:ring-slate-600',
+      'hover:border-slate-700 hover:bg-slate-700',
+    ].join(' '),
+  },
+  primary: {
+    outlined: [
+      'border-primary text-primary',
+      'focus:border-primary focus:ring-primary',
+      'hover:border-primary hover:bg-primary',
+    ].join(' '),
+    solid: [
+      'border-primary bg-primary text-white',
+      'focus:border-primary focus:ring-primary',
+      'hover:border-primary-dark hover:bg-primary-dark',
+    ].join(' '),
+  },
+  secondary: {
+    outlined: [
+      'border-secondary text-secondary',
+      'focus:border-secondary focus:ring-secondary',
+      'hover:border-secondary hover:bg-secondary',
+    ].join(' '),
+    solid: [
+      'border-secondary bg-secondary text-white',
+      'focus:border-secondary focus:ring-secondary',
+      'hover:border-secondary-dark hover:bg-secondary-dark',
+    ].join(' '),
+  },
+  accent: {
+    outlined: [
+      'border-accent text-accent',
+      'focus:border-accent focus:ring-accent',
+      'hover:border-accent hover:bg-accent',
+    ].join(' '),
+    solid: [
+      'border-accent bg-accent text-white',
+      'focus:border-accent focus:ring-accent',
+      'hover:border-accent-dark hover:bg-accent-dark',
+    ].join(' '),
+  },
+};
+
+const APPEARANCES = {
+  solid: 'solid',
+  outlined: 'outlined',
+};
+
+const SIZES = {
+  xs: 'text-xs px-2 py-0',
+  sm: 'text-sm px-3 py-0 h-6',
+  base: 'px-7 py-1 h-8 min-h-[2rem]',
+  lg: 'text-lg px-8 py-2 h-10 min-h-[2.5rem]',
+  xl: 'text-xl px-10 py-3 h-12 min-h-[3rem]',
+};
+
+function Button({
+  appearance,
+  busy,
+  children,
+  className,
+  color,
+  disabled,
+  onClick,
+  size,
+}) {
+  if (busy) {
+    disabled = true;
+  }
+
+  return (
+    <button
+      className={clsx(
+        'flex w-fit cursor-pointer items-center justify-center rounded-full',
```

Comment:
**todo**: on mobile clicking is selecting the text so I think adding [select-none](https://tailwindcss.com/docs/user-select#header) would be a nice default.

Reply:
122f9f6624656759b7e63690c8056f838161cc94